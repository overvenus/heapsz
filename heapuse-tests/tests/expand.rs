use std::{fs::read_dir, path::Path};

#[test]
fn expand_examples() {
    let current_path = snapbox::current_dir!();
    println!("current_path: {}", current_path.display());
    let example_dir = current_path.join("../../heapuse/examples");
    let dir = read_dir(example_dir).unwrap();
    for entry in dir {
        let entry = entry.unwrap();
        let mut name = entry.file_name().into_string().unwrap();
        if name.ends_with(".rs") {
            name.truncate(name.len() - 3);
            let expected = snapbox::Data::read_from(
                Path::new(&format!(
                    "{}/expand/{}.stdout",
                    current_path.display(),
                    &name
                )),
                None,
            );
            // cargo expand path::to::module
            snapbox::cmd::Command::new("cargo")
                .args(["expand", "-p", "heapuse", "--example", &name])
                .assert()
                .success()
                .stdout_matches(expected);
        }
    }
}
