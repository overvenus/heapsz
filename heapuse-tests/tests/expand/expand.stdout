#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
pub mod primitive {
    pub mod full {
        pub struct StructBool {
            #[heap_size]
            pub field_bool: bool,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructBool {
            #[inline]
            fn default() -> StructBool {
                StructBool {
                    field_bool: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructBool {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_bool)
            }
        }
        pub struct StructPrimitives {
            #[heap_size]
            pub field_bool: bool,
            #[heap_size]
            pub field_u8: u8,
            #[heap_size]
            pub field_i8: i8,
            #[heap_size]
            pub field_u16: u16,
            #[heap_size]
            pub field_i16: i16,
            #[heap_size]
            pub field_u32: u32,
            #[heap_size]
            pub field_i32: i32,
            #[heap_size]
            pub field_u64: u64,
            #[heap_size]
            pub field_i64: i64,
            #[heap_size]
            pub field_u128: u128,
            #[heap_size]
            pub field_i128: i128,
            #[heap_size]
            pub field_usize: usize,
            #[heap_size]
            pub field_isize: isize,
            #[heap_size]
            pub field_f32: f32,
            #[heap_size]
            pub field_f64: f64,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructPrimitives {
            #[inline]
            fn default() -> StructPrimitives {
                StructPrimitives {
                    field_bool: ::core::default::Default::default(),
                    field_u8: ::core::default::Default::default(),
                    field_i8: ::core::default::Default::default(),
                    field_u16: ::core::default::Default::default(),
                    field_i16: ::core::default::Default::default(),
                    field_u32: ::core::default::Default::default(),
                    field_i32: ::core::default::Default::default(),
                    field_u64: ::core::default::Default::default(),
                    field_i64: ::core::default::Default::default(),
                    field_u128: ::core::default::Default::default(),
                    field_i128: ::core::default::Default::default(),
                    field_usize: ::core::default::Default::default(),
                    field_isize: ::core::default::Default::default(),
                    field_f32: ::core::default::Default::default(),
                    field_f64: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructPrimitives {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_bool)
                    + ::heapuse::HeapSize::heap_size(&self.field_u8)
                    + ::heapuse::HeapSize::heap_size(&self.field_i8)
                    + ::heapuse::HeapSize::heap_size(&self.field_u16)
                    + ::heapuse::HeapSize::heap_size(&self.field_i16)
                    + ::heapuse::HeapSize::heap_size(&self.field_u32)
                    + ::heapuse::HeapSize::heap_size(&self.field_i32)
                    + ::heapuse::HeapSize::heap_size(&self.field_u64)
                    + ::heapuse::HeapSize::heap_size(&self.field_i64)
                    + ::heapuse::HeapSize::heap_size(&self.field_u128)
                    + ::heapuse::HeapSize::heap_size(&self.field_i128)
                    + ::heapuse::HeapSize::heap_size(&self.field_usize)
                    + ::heapuse::HeapSize::heap_size(&self.field_isize)
                    + ::heapuse::HeapSize::heap_size(&self.field_f32)
                    + ::heapuse::HeapSize::heap_size(&self.field_f64)
            }
        }
    }
    pub mod all {
        #[heap_size]
        pub struct StructBool {
            pub field_bool: bool,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructBool {
            #[inline]
            fn default() -> StructBool {
                StructBool {
                    field_bool: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructBool {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_bool)
            }
        }
        #[heap_size]
        pub struct StructPrimitives {
            pub field_bool: bool,
            pub field_u8: u8,
            pub field_i8: i8,
            pub field_u16: u16,
            pub field_i16: i16,
            pub field_u32: u32,
            pub field_i32: i32,
            pub field_u64: u64,
            pub field_i64: i64,
            pub field_u128: u128,
            pub field_i128: i128,
            pub field_usize: usize,
            pub field_isize: isize,
            pub field_f32: f32,
            pub field_f64: f64,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructPrimitives {
            #[inline]
            fn default() -> StructPrimitives {
                StructPrimitives {
                    field_bool: ::core::default::Default::default(),
                    field_u8: ::core::default::Default::default(),
                    field_i8: ::core::default::Default::default(),
                    field_u16: ::core::default::Default::default(),
                    field_i16: ::core::default::Default::default(),
                    field_u32: ::core::default::Default::default(),
                    field_i32: ::core::default::Default::default(),
                    field_u64: ::core::default::Default::default(),
                    field_i64: ::core::default::Default::default(),
                    field_u128: ::core::default::Default::default(),
                    field_i128: ::core::default::Default::default(),
                    field_usize: ::core::default::Default::default(),
                    field_isize: ::core::default::Default::default(),
                    field_f32: ::core::default::Default::default(),
                    field_f64: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructPrimitives {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_bool)
                    + ::heapuse::HeapSize::heap_size(&self.field_u8)
                    + ::heapuse::HeapSize::heap_size(&self.field_i8)
                    + ::heapuse::HeapSize::heap_size(&self.field_u16)
                    + ::heapuse::HeapSize::heap_size(&self.field_i16)
                    + ::heapuse::HeapSize::heap_size(&self.field_u32)
                    + ::heapuse::HeapSize::heap_size(&self.field_i32)
                    + ::heapuse::HeapSize::heap_size(&self.field_u64)
                    + ::heapuse::HeapSize::heap_size(&self.field_i64)
                    + ::heapuse::HeapSize::heap_size(&self.field_u128)
                    + ::heapuse::HeapSize::heap_size(&self.field_i128)
                    + ::heapuse::HeapSize::heap_size(&self.field_usize)
                    + ::heapuse::HeapSize::heap_size(&self.field_isize)
                    + ::heapuse::HeapSize::heap_size(&self.field_f32)
                    + ::heapuse::HeapSize::heap_size(&self.field_f64)
            }
        }
    }
    pub mod skip {
        #[heap_size]
        pub struct StructBool {
            #[heap_size(skip)]
            pub field_bool: bool,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructBool {
            #[inline]
            fn default() -> StructBool {
                StructBool {
                    field_bool: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructBool {
            fn heap_size(&self) -> usize {
                0
            }
        }
        #[heap_size]
        pub struct StructPrimitives {
            #[heap_size(skip)]
            pub field_bool: bool,
            #[heap_size(skip)]
            pub field_u8: u8,
            #[heap_size(skip)]
            pub field_i8: i8,
            #[heap_size(skip)]
            pub field_u16: u16,
            #[heap_size(skip)]
            pub field_i16: i16,
            #[heap_size(skip)]
            pub field_u32: u32,
            #[heap_size(skip)]
            pub field_i32: i32,
            #[heap_size(skip)]
            pub field_u64: u64,
            #[heap_size(skip)]
            pub field_i64: i64,
            #[heap_size(skip)]
            pub field_u128: u128,
            #[heap_size(skip)]
            pub field_i128: i128,
            #[heap_size(skip)]
            pub field_usize: usize,
            #[heap_size(skip)]
            pub field_isize: isize,
            #[heap_size(skip)]
            pub field_f32: f32,
            #[heap_size(skip)]
            pub field_f64: f64,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructPrimitives {
            #[inline]
            fn default() -> StructPrimitives {
                StructPrimitives {
                    field_bool: ::core::default::Default::default(),
                    field_u8: ::core::default::Default::default(),
                    field_i8: ::core::default::Default::default(),
                    field_u16: ::core::default::Default::default(),
                    field_i16: ::core::default::Default::default(),
                    field_u32: ::core::default::Default::default(),
                    field_i32: ::core::default::Default::default(),
                    field_u64: ::core::default::Default::default(),
                    field_i64: ::core::default::Default::default(),
                    field_u128: ::core::default::Default::default(),
                    field_i128: ::core::default::Default::default(),
                    field_usize: ::core::default::Default::default(),
                    field_isize: ::core::default::Default::default(),
                    field_f32: ::core::default::Default::default(),
                    field_f64: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructPrimitives {
            fn heap_size(&self) -> usize {
                0
            }
        }
    }
    pub mod blank {
        pub struct StructBool {
            pub field_bool: bool,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructBool {
            #[inline]
            fn default() -> StructBool {
                StructBool {
                    field_bool: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructBool {
            fn heap_size(&self) -> usize {
                0
            }
        }
        pub struct StructPrimitives {
            pub field_bool: bool,
            pub field_u8: u8,
            pub field_i8: i8,
            pub field_u16: u16,
            pub field_i16: i16,
            pub field_u32: u32,
            pub field_i32: i32,
            pub field_u64: u64,
            pub field_i64: i64,
            pub field_u128: u128,
            pub field_i128: i128,
            pub field_usize: usize,
            pub field_isize: isize,
            pub field_f32: f32,
            pub field_f64: f64,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructPrimitives {
            #[inline]
            fn default() -> StructPrimitives {
                StructPrimitives {
                    field_bool: ::core::default::Default::default(),
                    field_u8: ::core::default::Default::default(),
                    field_i8: ::core::default::Default::default(),
                    field_u16: ::core::default::Default::default(),
                    field_i16: ::core::default::Default::default(),
                    field_u32: ::core::default::Default::default(),
                    field_i32: ::core::default::Default::default(),
                    field_u64: ::core::default::Default::default(),
                    field_i64: ::core::default::Default::default(),
                    field_u128: ::core::default::Default::default(),
                    field_i128: ::core::default::Default::default(),
                    field_usize: ::core::default::Default::default(),
                    field_isize: ::core::default::Default::default(),
                    field_f32: ::core::default::Default::default(),
                    field_f64: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructPrimitives {
            fn heap_size(&self) -> usize {
                0
            }
        }
    }
    pub use full::StructBool;
    pub use full::StructPrimitives;
}
pub mod container {
    pub mod full {
        pub struct StructHeapMap {
            #[heap_size]
            pub field_map: std::collections::HashMap<
                usize,
                crate::primitive::StructBool,
            >,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructHeapMap {
            #[inline]
            fn default() -> StructHeapMap {
                StructHeapMap {
                    field_map: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructHeapMap {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_map)
            }
        }
        pub struct StructVec {
            #[heap_size]
            pub field_vec: Vec<crate::primitive::StructPrimitives>,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructVec {
            #[inline]
            fn default() -> StructVec {
                StructVec {
                    field_vec: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructVec {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_vec)
            }
        }
        pub struct StructContainers {
            #[heap_size]
            pub field_vec: Vec<crate::primitive::StructPrimitives>,
            #[heap_size]
            pub field_map: std::collections::HashMap<
                usize,
                crate::primitive::StructBool,
            >,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructContainers {
            #[inline]
            fn default() -> StructContainers {
                StructContainers {
                    field_vec: ::core::default::Default::default(),
                    field_map: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructContainers {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_vec)
                    + ::heapuse::HeapSize::heap_size(&self.field_map)
            }
        }
    }
    pub mod all {
        #[heap_size]
        pub struct StructHeapMap {
            pub field_map: std::collections::HashMap<
                usize,
                crate::primitive::StructBool,
            >,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructHeapMap {
            #[inline]
            fn default() -> StructHeapMap {
                StructHeapMap {
                    field_map: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructHeapMap {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_map)
            }
        }
        #[heap_size]
        pub struct StructVec {
            pub field_vec: Vec<crate::primitive::StructPrimitives>,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructVec {
            #[inline]
            fn default() -> StructVec {
                StructVec {
                    field_vec: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructVec {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_vec)
            }
        }
        #[heap_size]
        pub struct StructContainers {
            pub field_vec: Vec<crate::primitive::StructPrimitives>,
            pub field_map: std::collections::HashMap<
                usize,
                crate::primitive::StructBool,
            >,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructContainers {
            #[inline]
            fn default() -> StructContainers {
                StructContainers {
                    field_vec: ::core::default::Default::default(),
                    field_map: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructContainers {
            fn heap_size(&self) -> usize {
                0 + ::heapuse::HeapSize::heap_size(&self.field_vec)
                    + ::heapuse::HeapSize::heap_size(&self.field_map)
            }
        }
    }
    pub mod skip {
        #[heap_size]
        pub struct StructHeapMap {
            #[heap_size(skip)]
            pub field_map: std::collections::HashMap<
                usize,
                crate::primitive::StructBool,
            >,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructHeapMap {
            #[inline]
            fn default() -> StructHeapMap {
                StructHeapMap {
                    field_map: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructHeapMap {
            fn heap_size(&self) -> usize {
                0
            }
        }
        #[heap_size]
        pub struct StructVec {
            #[heap_size(skip)]
            pub field_vec: Vec<crate::primitive::StructPrimitives>,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructVec {
            #[inline]
            fn default() -> StructVec {
                StructVec {
                    field_vec: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructVec {
            fn heap_size(&self) -> usize {
                0
            }
        }
        #[heap_size]
        pub struct StructContainers {
            #[heap_size(skip)]
            pub field_vec: Vec<crate::primitive::StructPrimitives>,
            #[heap_size(skip)]
            pub field_map: std::collections::HashMap<
                usize,
                crate::primitive::StructBool,
            >,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructContainers {
            #[inline]
            fn default() -> StructContainers {
                StructContainers {
                    field_vec: ::core::default::Default::default(),
                    field_map: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructContainers {
            fn heap_size(&self) -> usize {
                0
            }
        }
    }
    pub mod blank {
        pub struct StructHeapMap {
            pub field_map: std::collections::HashMap<
                usize,
                crate::primitive::StructBool,
            >,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructHeapMap {
            #[inline]
            fn default() -> StructHeapMap {
                StructHeapMap {
                    field_map: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructHeapMap {
            fn heap_size(&self) -> usize {
                0
            }
        }
        pub struct StructVec {
            pub field_vec: Vec<crate::primitive::StructPrimitives>,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructVec {
            #[inline]
            fn default() -> StructVec {
                StructVec {
                    field_vec: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructVec {
            fn heap_size(&self) -> usize {
                0
            }
        }
        pub struct StructContainers {
            pub field_vec: Vec<crate::primitive::StructPrimitives>,
            pub field_map: std::collections::HashMap<
                usize,
                crate::primitive::StructBool,
            >,
        }
        #[automatically_derived]
        impl ::core::default::Default for StructContainers {
            #[inline]
            fn default() -> StructContainers {
                StructContainers {
                    field_vec: ::core::default::Default::default(),
                    field_map: ::core::default::Default::default(),
                }
            }
        }
        impl ::heapuse::HeapSize for StructContainers {
            fn heap_size(&self) -> usize {
                0
            }
        }
    }
    pub use full::StructHeapMap;
    pub use full::StructVec;
    pub use full::StructContainers;
}
mod unit_tuple {
    struct StructUnit;
    #[automatically_derived]
    impl ::core::default::Default for StructUnit {
        #[inline]
        fn default() -> StructUnit {
            StructUnit {}
        }
    }
    impl ::heapuse::HeapSize for StructUnit {
        fn heap_size(&self) -> usize {
            0
        }
    }
    struct StructTuple1(
        crate::primitive::StructBool,
        crate::primitive::StructPrimitives,
    );
    #[automatically_derived]
    impl ::core::default::Default for StructTuple1 {
        #[inline]
        fn default() -> StructTuple1 {
            StructTuple1(
                ::core::default::Default::default(),
                ::core::default::Default::default(),
            )
        }
    }
    impl ::heapuse::HeapSize for StructTuple1 {
        fn heap_size(&self) -> usize {
            0
        }
    }
    #[heap_size]
    struct StructTuple2(
        crate::primitive::StructBool,
        #[heap_size(skip)]
        crate::primitive::StructPrimitives,
    );
    #[automatically_derived]
    impl ::core::default::Default for StructTuple2 {
        #[inline]
        fn default() -> StructTuple2 {
            StructTuple2(
                ::core::default::Default::default(),
                ::core::default::Default::default(),
            )
        }
    }
    impl ::heapuse::HeapSize for StructTuple2 {
        fn heap_size(&self) -> usize {
            0 + ::heapuse::HeapSize::heap_size(&self.0)
        }
    }
}
pub mod with_attr {
    use self::remote::Inner;
    pub mod remote {
        pub struct Inner(pub Box<u8>);
        #[automatically_derived]
        impl ::core::default::Default for Inner {
            #[inline]
            fn default() -> Inner {
                Inner(::core::default::Default::default())
            }
        }
    }
    pub mod remote_inner_heap_size {
        use heapuse::HeapSize;
        use super::remote;
        pub fn heap_size(t: &remote::Inner) -> usize {
            t.0.heap_size()
        }
    }
    pub struct Wtih {
        #[heap_size(with = "remote_inner_heap_size")]
        pub inner: Inner,
    }
    #[automatically_derived]
    impl ::core::default::Default for Wtih {
        #[inline]
        fn default() -> Wtih {
            Wtih {
                inner: ::core::default::Default::default(),
            }
        }
    }
    impl ::heapuse::HeapSize for Wtih {
        fn heap_size(&self) -> usize {
            0 + remote_inner_heap_size::heap_size(&self.inner)
        }
    }
}
pub mod enums {
    use super::with_attr::{remote::*, *};
    pub enum Enum {
        A(#[heap_size] usize),
        B(#[heap_size] usize, usize),
        C(#[heap_size(with = "remote_inner_heap_size")] Inner, usize),
        D {
            #[heap_size(with = "remote_inner_heap_size")]
            inner: Inner,
            #[heap_size]
            size1: usize,
            size2: usize,
        },
    }
    impl ::heapuse::HeapSize for Enum {
        fn heap_size(&self) -> usize {
            #[allow(unused_variables)]
            match self {
                Self::A(f_0) => 0 + ::heapuse::HeapSize::heap_size(f_0),
                Self::B(f_0, f_1) => 0 + ::heapuse::HeapSize::heap_size(f_0),
                Self::C(f_0, f_1) => 0 + remote_inner_heap_size::heap_size(f_0),
                Self::D { inner, size1, size2 } => {
                    0 + remote_inner_heap_size::heap_size(inner)
                        + ::heapuse::HeapSize::heap_size(size1)
                }
            }
        }
    }
    #[heap_size]
    pub enum EnumAll {
        A(usize),
        B(usize, usize),
        C(#[heap_size(with = "remote_inner_heap_size")] Inner, usize),
        D {
            #[heap_size(with = "remote_inner_heap_size")]
            inner: Inner,
            size1: usize,
            size2: usize,
        },
    }
    impl ::heapuse::HeapSize for EnumAll {
        fn heap_size(&self) -> usize {
            #[allow(unused_variables)]
            match self {
                Self::A(f_0) => 0 + ::heapuse::HeapSize::heap_size(f_0),
                Self::B(f_0, f_1) => {
                    0 + ::heapuse::HeapSize::heap_size(f_0)
                        + ::heapuse::HeapSize::heap_size(f_1)
                }
                Self::C(f_0, f_1) => {
                    0 + remote_inner_heap_size::heap_size(f_0)
                        + ::heapuse::HeapSize::heap_size(f_1)
                }
                Self::D { inner, size1, size2 } => {
                    0 + remote_inner_heap_size::heap_size(inner)
                        + ::heapuse::HeapSize::heap_size(size1)
                        + ::heapuse::HeapSize::heap_size(size2)
                }
            }
        }
    }
    #[heap_size]
    pub enum EnumSkip {
        A(#[heap_size(skip)] Inner, usize),
        B { #[heap_size(skip)] inner: Inner, size1: usize, size2: usize },
        #[heap_size(skip)]
        C { inner: Inner },
    }
    impl ::heapuse::HeapSize for EnumSkip {
        fn heap_size(&self) -> usize {
            #[allow(unused_variables)]
            match self {
                Self::A(f_0, f_1) => 0 + ::heapuse::HeapSize::heap_size(f_1),
                Self::B { inner, size1, size2 } => {
                    0 + ::heapuse::HeapSize::heap_size(size1)
                        + ::heapuse::HeapSize::heap_size(size2)
                }
                Self::C { inner } => 0,
            }
        }
    }
}
